src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -176,7 +176,7 @@ private void OnHostingHttpRequestInStart(object arg)
 
         private void OnMvcBeforeAction(object arg)
         {
-            var httpContext = (HttpContext)BeforeActionHttpContextFetcher.Fetch(arg);
+            var httpContext = BeforeActionHttpContextFetcher.Fetch<HttpContext>(arg);
```

Comment:
Is this change (generic method vs casting) only cosmetic? It's fine if it is. I'm just curious.

Reply:
I changed the Fetch method overall to be generic to avoid boxing when returning a value type